### ISSUE TITLE
Proposal: @EntityData(withBuilder=true, callSuper=true)

### ISSUE THREAD

**Issue by Hervian** [rzwitserloot/lombok#1531]
# Change request
I use Lombok on @Entity annotated classes (i.e. classes used by a JPA ORM framework) and all my entities extend from a base class which adds an ID to the given entity.

As a result I end up with having (/needing) all of the following annotations on my entities:
```java
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@EqualsAndHashCode(callSuper = true)
@Data
```

This removes a lot of the cool development speed. **I think it would be very nice to have a super annotation**, something a la `@EntityData(withBuilder=true, callSuper=true)`.

## Elaboration
I like to have a builder for each entity. The @Builder annotation creates a constructor with a parameter for each field. But since the entities must have a non-null constructor I end up with the strange mix of @NoArgsConstructor and @AllArgsConstructor (inspired by [this post](https://stackoverflow.com/a/34299054/6095334)).

**Comment by Maaartinus** [comment]
@Hervian In case you think that this can be generalized as #557, you may want to close this as duplicate.

**Comment by Hervian** [comment]
At @Maaartinus  points out, this change request would be solved/implemented by [#557](https://github.com/rzwitserloot/lombok/issues/557). 

This would at least be the case if emphasis is put on the following line from that issue's description: 
*"@ AllArgsConstructor is definitely not the only annotation that might make sense on".*

That is, if the annotations mentioned in this issue, was also modified to be legal to use on annotation types (in addition to AllArgsConstructor).

**Comment by Hervian** [comment]
#557
